app/scripts/lib/util.ts

```diff
@@ -480,3 +481,138 @@ export function getConversionRatesForNativeAsset({
+    try {
+      const chainsList = await getSafeChainsList();
+      knownDomainsSet = new Set<string>();
+
+      for (const chain of chainsList) {
+        if (chain.rpc && Array.isArray(chain.rpc)) {
+          for (const rpcUrl of chain.rpc) {
+            try {
+              // Handle placeholder variables like ${INFURA_API_KEY}
+              const cleanedUrl = rpcUrl.replace(/\$\{[^}]+\}/gu, 'API_KEY');
+              const url = new URL(cleanedUrl);
+              knownDomainsSet.add(url.hostname);
+            } catch (e) {
+              // Skip invalid URLs
+              continue;
+            }
+          }
+        }
+      }
+    } catch (error) {
+      console.error('Error initializing known domains:', error);
+      knownDomainsSet = new Set<string>();
+    }
+  })();
+
+  return initPromise;
+}
+
+/**
+ * Check if a domain is in the known domains list
+ *
+ * @param domain - The domain to check
+ */
+export function isKnownDomain(domain: string): boolean {
+  return knownDomainsSet?.has(domain) ?? false;
+}
+
+/**
+ * Extracts the domain from an RPC endpoint URL with privacy considerations
+ *
+ * @param rpcUrl - The RPC endpoint URL
+ * @param knownDomainsForTesting - Optional Set of known domains for testing purposes
+ * @returns The domain for known providers, 'private' for private/custom networks, or 'invalid' for invalid URLs
+ */
+export function extractRpcDomain(
+  rpcUrl: string,
+  knownDomainsForTesting?: Set<string>,
+): string {
+  if (!rpcUrl) {
+    return 'invalid';
+  }
+
+  const trimmedRpcUrl = rpcUrl.trim();
+
+  try {
+    let url;
+    try {
+      url = new URL(trimmedRpcUrl);
+    } catch (e) {
+      if (
+        !trimmedRpcUrl.startsWith('http://') &&
+        !trimmedRpcUrl.startsWith('https://')
+      ) {
+        try {
+          url = new URL(`https://${trimmedRpcUrl}`);
+        } catch (e2) {
+          return 'invalid';
+        }
+      } else {
+        return 'invalid';
+      }
+    }
+
+    const hostname = url.hostname.toLowerCase();
+
+    if (
+      hostname === 'localhost' ||
+      hostname === '::1' ||
+      hostname.startsWith('127.')
+    ) {
+      return 'private';
+    }
+
+    const validIPv4 =
+      /^(?:(?:25[0-5]|2[0-4]\d|1?\d{1,2})\.){3}(?:25[0-5]|2[0-4]\d|1?\d{1,2})$/u;
+
+    const isPrivateIPv4 = (ip: string): boolean => {
+      if (!validIPv4.test(ip)) {
+        return false;
+      }
+      const [a, b] = ip.split('.').map(Number);
+      return (
+        a === 10 ||
+        (a === 172 && b >= 16 && b <= 31) ||
+        (a === 192 && b === 168)
+      );
+    };
+
+    if (validIPv4.test(hostname) && isPrivateIPv4(hostname)) {
+      return 'private';
+    }
+
+    if (hostname.includes(':')) {
+      return 'private';
+    }
+
+    if (hostname.startsWith('xn--')) {
+      return 'private';
+    }
+
+    const domainsToCheck = knownDomainsForTesting || knownDomainsSet;
+    if (domainsToCheck?.has(hostname)) {
+      return hostname;
+    }
```

Comment:
If you put this check _first_ I think you can get rid of the rest of the logic and return `private` in the `else` case. Previously the code was trying to differentiate between different types of urls, but now that we only have two it doesn't make much sense to do the more complicated logic above.

Reply:
Yeah I rushed that change. Sorry. I got it updated and ran the tests.